React/Views/RCTScrollView.m

```diff
@@ -317,8 +317,8 @@ - (void)setFrame:(CGRect)frame
   CGSize boundsSize = self.bounds.size;
 
   self.contentOffset = CGPointMake(
-    MAX(0, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
-    MAX(0, MIN(originalOffset.y, fullContentSize.height - boundsSize.height)));
+    MAX(-contentInset.left, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
```

Comment:
Having reactive and declarative nature of React and React Native, I don't think that even having `initialContentOffset` will make it clear. Because... Which moment is "initial"? What if some ancestor component was rebuilt? How RN should response on changing this value later, on runtime?

@sahrens Do you have any thoughts about it?